### PR TITLE
general: better solution for application initialization

### PR DESCRIPTION
**Pull request by Garfield-fr**
Co-Authored-by: Bertrand Zuchuat <bertrand.zuchuat@rero.ch>

## Why are you opening this PR?

- To implement better solution for application initialization

## How to test?

- Check application

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?
- [ ] Extracted translations?

### CODE REVIEW

File: projects/admin/src/app/service/app-init.service.ts [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+            if (this._userService.allowAccess) {
+              this._organisationService.loadOrganisationByPid(
+                data.metadata.library.organisation.pid
+              );
+              this.initializeLocaleStorage();
+            }
+          }
+        });
+        resolve();
+      }, 200);
```
Comment (sebastiendeleze) [inline projects/admin/src/app/service/app-init.service.ts]: Is it necessary to wait 200ms?

File: projects/admin/src/app/service/user.service.ts [hunk 2]
```diff
@@ -86,4 +129,17 @@ export class UserService {
...
   }
+
+  /**
+   * Check if you are an access ton admin interface
+   */
+  private isAuthorizedAccess() {
+    const adminRoles = this._appConfigService.adminRoles;
+    this._allowInterfaceAccess = this.user.getRoles()
+    .filter((role: string) => {
+      if (adminRoles.indexOf(role) > -1) {
```
Comment (jma) [inline projects/admin/src/app/service/user.service.ts]: You can probably use some here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some

File: projects/admin/src/app/app.component.html [hunk 3]
```diff
@@ -14,12 +14,12 @@
   You should have received a copy of the GNU Affero General Public License
   along with this program.  If not, see <http://www.gnu.org/licenses/>.
 -->
-<div *ngIf="!user" translate>Loading…</div>
-<ng-container *ngIf="user">
+<div *ngIf="!userLoaded" translate>Loading…</div>
```
Comment (jma) [inline projects/admin/src/app/app.component.html]: We can use a ng-template with an else condition.

File: projects/admin/src/app/class/user.ts [hunk 4]
```diff
@@ -54,15 +54,11 @@ export class User {
   }
 
   /**
-   * Check if you are an access ton admin interface
-   * @param roles - arrays of role
+   * Get Roles
+   * @return array
```
Comment (jma) [inline projects/admin/src/app/class/user.ts]: array of roles?

File: projects/admin/src/app/class/user.ts [hunk 5]
```diff
@@ -54,15 +54,11 @@ export class User {
...
    */
-  isAuthorizedAdminAccess(roles: Array<string>) {
-    return this.roles.filter((role: string) => {
-      if (roles.indexOf(role) > -1) {
-        return role;
-      }
-    }).length > 0;
+  getRoles() {
+    return this.roles;
```
Comment (jma) [inline projects/admin/src/app/class/user.ts]: Is a getter makes sense for a public property?

File: projects/admin/src/app/service/app-init.service.ts [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+    private _translateService: TranslateService,
+    private _organisationService: OrganisationService,
+    private _localeStorageService: LocalStorageService,
+    private _librarySwitchService: LibrarySwitchService
+  ) { }
+
+  /** Init */
+  init() {
+    return new Promise<void>((resolve) => {
+      setTimeout(() => {
```
Comment (jma) [inline projects/admin/src/app/service/app-init.service.ts]: Can you put a comment to explain why this is required? Look like a hack.

File: projects/admin/src/app/service/user.service.ts [hunk 7]
```diff
@@ -21,56 +21,99 @@ import { RecordService } from '@rero/ng-core';
...
 
 @Injectable({
   providedIn: 'root'
 })
 export class UserService {
 
+  /** Event after user loaded */
   private onUserLoaded: Subject<User> = new Subject();
 
+  /** User object */
```
Comment (jma) [inline projects/admin/src/app/service/user.service.ts]: _ is missing for private property

**Pull request closed without merging**